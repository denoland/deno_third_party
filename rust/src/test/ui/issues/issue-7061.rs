struct BarStruct;

impl<'a> BarStruct {
    fn foo(&'a mut self) -> Box<BarStruct> { self }
    //~^ ERROR mismatched types
    //~| expected type `std::boxed::Box<BarStruct>`
    //~| found type `&'a mut BarStruct`
}

fn main() {}
