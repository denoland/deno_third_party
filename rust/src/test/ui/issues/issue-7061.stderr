error[E0308]: mismatched types
  --> $DIR/issue-7061.rs:4:46
   |
LL |     fn foo(&'a mut self) -> Box<BarStruct> { self }
   |                             --------------   ^^^^ expected struct `std::boxed::Box`, found mutable reference
   |                             |
   |                             expected `std::boxed::Box<BarStruct>` because of return type
   |
   = note: expected type `std::boxed::Box<BarStruct>`
              found type `&'a mut BarStruct`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
